build-tools/packages/build-cli/src/library/packageExports.ts

```diff
@@ -244,3 +244,134 @@ export function queryTypesResolutionPathsFromPackageExports<TOutKey>(
 
 	return { mapKeyToOutput, mapNode10CompatExportPathToData, mapTypesPathToExportPaths };
 }
+
+/**
+ * Read package "exports" to determine which of given "default" paths are present in map.
+ *
+ * @param packageJson - json content of package.json
+ * @param mapSrcQueryPathToOutKey - keys of map represent paths to match.
+ * @param emitDeclarationOnly - If true, "types" exports are considered.
+ * @param logger - optional Logger
+ * @returns object with mapKeyToOutput.
+ */
+export function queryDefaultResolutionPathsFromPackageExports<TOutKey>(
```

Comment:
This function seems like it is replicating the exports resolution algorithm. We can just use that resolution logic directly. See use of [resolve.imports](https://www.npmjs.com/package/resolve.exports) elsewhere.

Reply:
Related to an earlier comment - it's confusing me when we read from the exports field if we are also reading from index.ts to find the exports. Maybe a high level bulleted list of the steps would help? I thought it was:

- Open the file passed into the CLI to find the full list of exports.
- Determine what API levels are exported using the data from the previous step.
- Create a TS file with the exports for the appropriate level.